diff --git a/src/database/mongo.js b/src/database/mongo.js
index 0e69c12..5f8aae2 100644
--- a/src/database/mongo.js
+++ b/src/database/mongo.js
@@ -22,12 +22,14 @@
 		},
 		{
 			name: 'mongo:username',
-			description: 'MongoDB username'
+			description: 'MongoDB username',
+			'default': nconf.get('mongo:username') || ''
 		},
 		{
 			name: 'mongo:password',
 			description: 'Password of your MongoDB database',
-			hidden: true
+			hidden: true,
+			before: function(value) { value = value || nconf.get('mongo:password') || ''; return value; }
 		},
 		{
 			name: "mongo:database",
diff --git a/src/database/redis.js b/src/database/redis.js
index 727705c..4c306ce 100644
--- a/src/database/redis.js
+++ b/src/database/redis.js
@@ -28,7 +28,8 @@
 		{
 			name: 'redis:password',
 			description: 'Password of your Redis database',
-			hidden: true
+			hidden: true,
+			before: function(value) { value = value || nconf.get('redis:password') || ''; return value; }
 		},
 		{
 			name: "redis:database",
diff --git a/src/install.js b/src/install.js
index 45d44d7..be802f5 100644
--- a/src/install.js
+++ b/src/install.js
@@ -47,7 +47,7 @@ questions.main = [
 questions.optional = [
 	{
 		name: 'port',
-		default: 4567
+		default: nconf.get('port') || 4567
 	}
 ];
 
